docs/Modeling-Artifacts.md

```diff
@@ -0,0 +1,378 @@
+
+`Copy(m0)` copies the graph rooted by the node `m0`, including `m0` itself and all of its successors `b0`, `b1`, and `b2`.
+
+```mermaid
+graph TD;
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+```
+
+`Copy(m2)` copies the graph rooted at the node `m2`, including `m2` itself, its successor `b5`, and the subgraph rooted at `m0`.
+
+```mermaid
+graph TD;
+
+M2["Manifest m2"]--config-->Blob0
+M2--layers-->Blob5["Blob b5"]
+M2--subject-->M0
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+```
+
+`Copy(b0)` copies itself only as it has no successor.
+
+```mermaid
+graph TD;
+
+Blob0["Blob b0"]
+```
+
+### Extended Copy
+
+As an extension to the `Copy` function, the `ExtendedCopy` function is designed to replicate the entire graph reachable from any given node in a DAG. This method requires that the source CAS supports predecessor finding (i.e., it indexes predecessor relationships when storing the graph).
+
+The predecessor relationship for the [example graph](#graph-concepts) looks like this:
+
+```mermaid
+graph TD;
+
+Blob0["Blob b0"]--predecessor-->M0["Manifest m0"]
+Blob1["Blob b1"]--predecessor-->M0
+Blob2["Blob b2"]--predecessor-->M0
+
+Blob3["Blob b3"]--predecessor-->M1["Manifest m1"]
+Blob4["Blob b4"]--predecessor-->M1
+
+Blob5["Blob b5"]--predecessor-->M2
+M0--predecessor-->M2
+Blob0--predecessor-->M2["Manifest m2"]
+
+M1--predecessor-->I0
+M0--predecessor-->I0["Index i0"]
+```
+
+With the predecessor finding capability, `ExtendedCopy` recursively calls `Predecessors()` to discover root nodes, then applies `Copy` on each discovered root. For instance:
+
+`ExtendedCopy(b5)` finds out the root node `m2` starting from `b5`, and copies the graph rooted at `m2`:
+
+```mermaid
+graph TD;
+
+M2["Manifest m2"]--config-->Blob0
+M2--layers-->Blob5["Blob b5"]
+M2--subject-->M0
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+```
+
+`ExtendedCopy(m1)` determines the root node `i0` from `m1`, and copies the graph rooted at `i0`:
+
+```mermaid
+graph TD;
+
+I0["Index i0"]--manifests-->M0
+I0--manifests-->M1
+
+M1["Manifest m1"]--config-->Blob3["Blob b3"]
+M1--layers-->Blob4["Blob b4"]
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+```
+
+`ExtendedCopy(b0)` finds multiple root nodes `m2` and `i0` starting from `b0`, then copies the combined graph rooted at  `m2` and `i0`:
+
+```mermaid
+graph TD;
+
+I0["Index i0"]--manifests-->M0
+I0--manifests-->M1
+
+M2["Manifest m2"]--config-->Blob0
+M2--layers-->Blob5["Blob b5"]
+M2--subject-->M0
+
+M1["Manifest m1"]--config-->Blob3["Blob b3"]
+M1--layers-->Blob4["Blob b4"]
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+```
+
+#### Referrers API / Referrers Tag Schema
```

Comment:
nit: should this be a subsection of `Extended Copy`? I think it can be at the same level of `Extended Copy` (### instead of ####)